.azure-pipelines/testcases_collection/report_data_storage.py

```diff
@@ -0,0 +1,63 @@
+"""Wrappers and utilities for storing test cases."""
+import json
+import os
+import tempfile
+
+from azure.kusto.data import KustoConnectionStringBuilder
+from azure.kusto.ingest import QueuedIngestClient as KustoIngestClient
+from azure.kusto.ingest import IngestionProperties
+from azure.kusto.data.data_format import DataFormat
+from datetime import datetime
+
+
+class KustoConnector():
+    """KustoReportDB is a wrapper for storing test reports in Kusto/Azure Data Explorer."""
+
+    TESTCASES_TABLE = "TestCasesCollection"
+
+    def __init__(self, db_name: str):
+        """Initialize a Kusto report DB connector.
+
+        Args:
+            db_name: The Kusto database to connect to.
+        """
+        self.db_name = db_name
+
+        ingest_cluster = os.getenv("TEST_REPORT_INGEST_KUSTO_CLUSTER_URL")
+        access_token = os.getenv('ACCESS_TOKEN', None)
+
+        if not ingest_cluster or not access_token:
+            raise RuntimeError(
+                "Could not load Kusto Credentials from environment")
+        else:
+            kcsb = KustoConnectionStringBuilder.with_aad_application_token_authentication(ingest_cluster, access_token)
+            self._ingestion_client = KustoIngestClient(kcsb)
+
+    def upload_testcases(self, test_cases):
+        uploadtime = str(datetime.now())
+
+        for case in test_cases:
+            case["uploadtime"] = uploadtime
+
+        print("Upload test scripts")
+        self._ingest_data(self.TESTCASES_TABLE, test_cases)
+
+    def _ingest_data(self, table, data):
+        props = IngestionProperties(
+            database=self.db_name,
+            table=table,
+            data_format=DataFormat.JSON,
+            ingestion_mapping_reference="TestCasesCollectionMapping"
```

Comment:
This KustoConnector looks like a generic purpose class. However, this hard code makes it only working for the TestCaseCollection table.

Reply:
Use a constant to record this value.